src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -227,7 +227,17 @@ private static Uri GetUriFromString(string value)
             else
             {
                 host = value.Substring(0, separatorIndex);
-                if (!ushort.TryParse(value.AsSpan(separatorIndex + 1), out port))
+                int endIndex = separatorIndex + 1;
```

Comment:
Why does this string parsing code exist at all? Why not just convert the string to a URI and then extract the host and port portions of it?

Reply:
It was like that at some point. It did not handle well cases when the value is for example just IP (or ip + port) The other issue was handling special characters in password. So we end up breaking it to fragments on our own.